src/PageContent/Calendar/DescriptionContent.js

```diff
@@ -0,0 +1,208 @@
+        });
+    }
+
+    addRow() {
+        this.edit();
+        let blankRow = { name: '', amount: 0, unit: '' };
+        let currentDonation = this.state.donationObject;
+        currentDonation.push(blankRow);
+        this.setState({
+            donationObject: currentDonation
+        });
+    }
+
+    stringifyDonation(donation) {
+        let foodList = '';
+        if (donation !== undefined) {
+            if (donation.length > 0) {
+                foodList +=
+                    donation[0].name +
+                    ' ' +
+                    donation[0].amount +
+                    ' ' +
+                    donation[0].unit;
+                for (let i = 1; i < donation.length; i++) {
+                    foodList +=
+                        ', ' +
+                        donation[i].name +
+                        ' ' +
+                        donation[i].amount +
+                        ' ' +
+                        donation[i].unit;
+                }
+            }
+        }
+        return foodList;
+    }
+
+    handleChange(e) {
+        e.preventDefault();
+        // pass these in from firebase
+        let newDonation = [];
+        for (let i = 0; i < this.state.donationObject.length; i++) {
+            let name = e.target[i + 'name'].value;
+            let amount = e.target[i + 'amount'].value;
+            let unit = e.target[i + 'unit'].value;
+            if (name !== '') {
+                newDonation.push({ name: name, amount: amount, unit: unit });
+            }
+        }
+        this.setState({
+            donationObject: newDonation,
+            edit: false,
+            editedBy: this.props.accountOwnerName,
+            editedAt: moment().format('MM/DD h:mma')
+        });
+
+        // TODO: push data to firebase
+    }
+
+    render() {
+        let donation = this.stringifyDonation(this.state.donationObject);
+
+        let editDonation = '';
+
+        if (this.state.donationObject.length > 0) {
+            editDonation = this.state.donationObject.map((item, index) => {
+                return (
+                    <div className="donation-edit-wrapper">
+                        <input
+                            type="text"
+                            className="food"
+                            defaultValue={item.name}
+                            name={index + 'name'}
+                        />
+                        <div className="weight-unit-wrapper">
+                            <input
+                                type="number"
+                                className="weight"
+                                defaultValue={item.amount}
+                                name={index + 'amount'}
+                            />
+                            <select
+                                className="description-unit"
+                                defaultValue={item.unit}
+                                name={index + 'unit'}
+                            >
```

Comment:
add a TODO for conditional validation: if name is entered, they need to enter a quantity > 0